From e41ca14a124d44a477272cac44605c46497c4cf4 Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Wed, 28 Jun 2017 18:33:16 +0800
Subject: [PATCH 2028/5242] MLK-15309-02 arm64: dts: imx8mq-evk: add the enet
 PHY led_act blinding workaround

commit  3e7188fca665a192a000ea61ea2854e22efe30b0 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add enet PHY AR8031 led_act blinding issue sw workaround.
Adjust the PIN drive strength for the better timing.
Set the PTP ref clock to 100Mhz.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Reviewed-by: Pandy Gao <pandy.gao@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts |   20 +++++++++++---------
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi    |    4 +++-
 2 files changed, 14 insertions(+), 10 deletions(-)
 mode change 100755 => 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
 mode change 100755 => 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
old mode 100755
new mode 100644
index 61523ff..dc40452
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -67,14 +67,15 @@
 				MX8MQ_IOMUXC_ENET_TD2_ENET1_RGMII_TD2	0x1f
 				MX8MQ_IOMUXC_ENET_TD1_ENET1_RGMII_TD1	0x1f
 				MX8MQ_IOMUXC_ENET_TD0_ENET1_RGMII_TD0	0x1f
+				MX8MQ_IOMUXC_ENET_RD3_ENET1_RGMII_RD3	0x91
+				MX8MQ_IOMUXC_ENET_RD2_ENET1_RGMII_RD2	0x91
+				MX8MQ_IOMUXC_ENET_RD1_ENET1_RGMII_RD1	0x91
+				MX8MQ_IOMUXC_ENET_RD0_ENET1_RGMII_RD0	0x91
 				MX8MQ_IOMUXC_ENET_TXC_ENET1_RGMII_TXC	0x1f
-				MX8MQ_IOMUXC_ENET_RD3_ENET1_RGMII_RD3	0x1f
-				MX8MQ_IOMUXC_ENET_RD2_ENET1_RGMII_RD2	0x1f
-				MX8MQ_IOMUXC_ENET_RD1_ENET1_RGMII_RD1	0x1f
-				MX8MQ_IOMUXC_ENET_RD0_ENET1_RGMII_RD0	0x1f
-				MX8MQ_IOMUXC_ENET_RXC_ENET1_RGMII_RXC	0x1f
-				MX8MQ_IOMUXC_ENET_RX_CTL_ENET1_RGMII_RX_CTL	0x1f
+				MX8MQ_IOMUXC_ENET_RXC_ENET1_RGMII_RXC	0x91
+				MX8MQ_IOMUXC_ENET_RX_CTL_ENET1_RGMII_RX_CTL	0x91
 				MX8MQ_IOMUXC_ENET_TX_CTL_ENET1_RGMII_TX_CTL	0x1f
+				MX8MQ_IOMUXC_GPIO1_IO09_GPIO1_IO9	0x19
 			>;
 		};
 
@@ -141,10 +142,9 @@
 
 &fec1 {
 	pinctrl-names = "default";
-	inctrl-0 = <&pinctrl_fec1>;
-	phy-mode = "rgmii";
+	pinctrl-0 = <&pinctrl_fec1>;
+	phy-mode = "rgmii-id";
 	phy-handle = <&ethphy0>;
-	fsl,ar8031-phy-fixup;
 	fsl,magic-packet;
 	status = "okay";
 
@@ -155,6 +155,8 @@
 		ethphy0: ethernet-phy@0 {
 			compatible = "ethernet-phy-ieee802.3-c22";
 			reg = <0>;
+			at803x,led-act-blind-workaround;
+			at803x,eee-disabled;
 		};
 	};
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
old mode 100755
new mode 100644
index 75530a71..bbda606
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -560,10 +560,12 @@
 			"enet_clk_ref", "enet_out";
 		assigned-clocks = <&clk IMX8MQ_CLK_ENET_AXI_SRC>,
 				  <&clk IMX8MQ_CLK_ENET_TIMER_SRC>,
-				  <&clk IMX8MQ_CLK_ENET_REF_SRC>;
+				  <&clk IMX8MQ_CLK_ENET_REF_SRC>,
+				  <&clk IMX8MQ_CLK_ENET_TIMER_DIV>;
 		assigned-clock-parents = <&clk IMX8MQ_SYS1_PLL_266M>,
 					 <&clk IMX8MQ_SYS2_PLL_100M>,
 					 <&clk IMX8MQ_SYS2_PLL_125M>;
+		assigned-clock-rates = <0>, <0>, <125000000>, <100000000>;
 		stop-mode = <&gpr 0x10 3>;
 		fsl,num-tx-queues=<3>;
 		fsl,num-rx-queues=<3>;
-- 
1.7.9.5

